#first case   =>   x = 3
#since x is initialized in the outer scope and accessible in an inner scope.
#(block immediately follows a method invocation)



#second case  =>   error, x not difined.
#since x is initialized in the inner scope and isn't accessible in an outer scope. 